services/skus/service_nonint_test.go

```diff
@@ -2695,6 +2696,365 @@ func TestShouldUpdateOrderStripeSubID(t *testing.T) {
 	}
 }
 
+func TestService_createRadomSessID(t *testing.T) {
+	type tcExpected struct {
+		sessionID string
+		mustErr   must.ErrorAssertionFunc
+	}
+
+	type tcGiven struct {
+		req   *model.CreateOrderRequestNew
+		order *model.Order
+		radCl radomClient
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "err_success_url",
+			given: tcGiven{
+				order: &model.Order{},
+				req: &model.CreateOrderRequestNew{
+					RadomMetadata: &model.OrderRadomMetadata{
+						SuccessURI: "https://invalid%.com",
+					},
+				},
+			},
+			exp: tcExpected{
+				mustErr: func(t must.TestingT, err error, i ...interface{}) {
+					must.NotNil(t, err)
+				},
```

Comment:
issue: This way of expressing the error has two issues: 1) it couples expectations with a specific library which [this way of expression](https://github.com/brave-intl/bat-go/blob/master/services/skus/service_nonint_test.go#L3899-L4063) does not do; and 2) it makes the expectations harder to read and comprehend, compared to [the same way](https://github.com/brave-intl/bat-go/blob/master/services/skus/service_nonint_test.go#L3899-L4063), where the logic of dealing with errors is expressed in the test code itself.

The given example (and that's how the majority of tests is and has been written now) shifts focus to the actual expectation (i.e. error), and deals with the handling as part of the testing code, only once. The proposed way does the opposite, which has a negative effect – it makes the expectation declarations noisy at the expense of simplifying the testing code by one line at most. I would much rather have simple declarations and deal with the consequences in the testing code. The examples are plentiful now across the entire SKUs service, and there has never been a case _yet_ where it would not work. I would bet if it does not, then the code that is being tested is not well expressed enough.

Reply:
We have discussed this, and after the conversation I see the benefits of using it in that particular instance.